core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -177,6 +177,16 @@ case object OnHeapUnifiedMemory extends MemoryManagerExecutorMetricType(
 case object OffHeapUnifiedMemory extends MemoryManagerExecutorMetricType(
   (m => m.offHeapExecutionMemoryUsed + m.offHeapStorageMemoryUsed))
 
+case object OnHeapUnUnifiedMemory extends MemoryManagerExecutorMetricType(
```

Comment:
why do we need this? `JVMHeapMemory` reports it already and people can easily calculate it by themselves.

Reply:
> why do we need this? `JVMHeapMemory` reports it already and people can easily calculate it by themselves.

Always we use event log/restful API to get the peakMemoryMetrics. what we can get is the peakJVMOnHeap/peakOnHeapStorageMemory/peakOnHeapExecution memory during a period of time.
But we can use the final status of peakJVMOnHeap/peakOnHeapStorageMemory/peakOnHeapExecution  to calculate  and get the peak unmanaged memory usage.

Reply:
where do we calculate the peak value? the framework?

Reply:
> where do we calculate the peak value? the framework?

Yea. 

```
case object JVMHeapMemory extends SingleValueExecutorMetricType {
  override private[spark] def getMetricValue(memoryManager: MemoryManager): Long = {
    ManagementFactory.getMemoryMXBean.getHeapMemoryUsage().getUsed()
  }
}
```

Reply:
Any more suggestion?